static/js/stream_ui_updates.js

```diff
@@ -162,6 +163,32 @@ export function update_permissions_banner(sub) {
     $settings.find(".stream-settings-tip-container").html(rendered_tip);
 }
 
+export function update_notifications_stream_in_settings() {
+    // if stream creation form is not open, return
+    if (!hash_util.is_create_new_stream_narrow()) {
+        return;
+    }
+
+    const notifications_stream = stream_data.get_notifications_stream();
+    const stream_privacy_symbol_html = render_stream_privacy({
+        invite_only: notifications_stream.invite_only,
+        is_web_public: notifications_stream.is_web_public,
+    });
+    const $replacement_html = $(
+        "<strong> " + stream_privacy_symbol_html + notifications_stream + "</strong>",
+    );
+
+    // replace old announcement stream with new announcement stream
+    $("#announce-new-stream .checkbox strong").replaceWith($replacement_html);
+
+    // if announcement is turned off, hide the checkbox
+    if (!stream_data.realm_has_notifications_stream()) {
+        $("#announce-new-stream .checkbox").hide();
```

Comment:
This check can be move at the top and we can return early after hiding if there is no notification stream.